x/emissions/proto/emissions/v6/tx.proto

```diff
@@ -0,0 +1,547 @@
+  repeated uint64 max_top_reputers_to_reward = 23;
+  repeated string create_topic_fee = 24 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  repeated uint64 gradient_descent_max_iters = 25;
+  repeated string registration_fee = 28 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  repeated uint64 default_page_limit = 29;
+  repeated uint64 max_page_limit = 30;
+  repeated int64 min_epoch_length_record_limit = 31;
+  repeated uint64 blocks_per_month = 32;
+  repeated string p_reward_inference = 33 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string p_reward_forecast = 34 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string p_reward_reputer = 35 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string c_reward_inference = 36 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string c_reward_forecast = 37 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string c_norm = 38 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string epsilon_reputer = 40 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated uint64 half_max_process_stake_removals_end_block = 42;
+  repeated string data_sending_fee = 43 [
+    (cosmos_proto.scalar) = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+  repeated string epsilon_safe_div = 44 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated uint64 max_elements_per_forecast = 45;
+  repeated uint64 max_active_topics_per_block = 46;
+  repeated uint64 max_string_length = 47;
+  repeated string initial_regret_quantile = 48 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated string p_norm_safe_div = 49 [
+    (gogoproto.customtype) = "github.com/allora-network/allora-chain/math.Dec",
+    (gogoproto.nullable) = false
+  ];
+  repeated bool global_whitelist_enabled = 50;
+  repeated bool topic_creator_whitelist_enabled = 51;
+}
+
+message UpdateParamsRequest {
+  option (cosmos.msg.v1.signer) = "sender";
+  string sender = 1;
```

Comment:
Also related to types checks mentioned in other comments, adding [proto scalars](https://docs.cosmos.network/v0.50/build/building-modules/protobuf-annotations#scalar) can give a hint to clients to enforce the type, but it's only on the client side..

```suggestion
  string sender = 1 string [(cosmos_proto.scalar) = "cosmos.AddressString"];
```

Reply:
Nice suggestion! And it won't require any migration.

Reply:
I think we can do this in a separate PR

Reply:
https://linear.app/alloralabs/issue/PROTO-2953/add-proto-scalars-to-all-proto-props